include/z64player.h

```diff
@@ -167,6 +167,78 @@ typedef enum PlayerItemAction {
     /* 0x53 */ PLAYER_IA_MAX
 } PlayerItemAction;
 
+#define GET_B_SWORD_FROM_AP(actionParam) ((actionParam) - PLAYER_IA_SWORD_KOKIRI + 1)
+
+typedef enum PlayerBButtonSword {
+    /* 0 */ PLAYER_BSWORD_NONE,
+    /* 1 */ PLAYER_BSWORD_KOKIRI = GET_B_SWORD_FROM_AP(PLAYER_IA_SWORD_KOKIRI),
+    /* 2 */ PLAYER_BSWORD_RAZOR = GET_B_SWORD_FROM_AP(PLAYER_IA_SWORD_RAZOR),
+    /* 3 */ PLAYER_BSWORD_GILDED = GET_B_SWORD_FROM_AP(PLAYER_IA_SWORD_GILDED),
+    /* 4 */ PLAYER_BSWORD_MAX
+} PlayerBButtonSword;
```

Comment:
I don't like this name. It may be accurate, but is this actually what it's looking for? How is it used? What is the OoT equivalent if any? Etc. Also the spacing is inconsistent between the macro and enum.

There is also the possibility it's just an initial segment of the next one.

Reply:
> Also the spacing is inconsistent between the macro and enum

What do you mean?

Reply:
This macro is used in `func_80124110` (renamed to `Player_BButtonSwordFromIA` in this PR)

That function is used in 2 places in the repo, as a wrapper in `func_80124148` (`Player_GetHeldBButtonSword`) and in `func_8082FE0C` of ovl_player_actor  (which also calls `Player_GetHeldBButtonSword`)


In [player](https://github.com/Decompollaborate/mm/blob/65f81484abbaa38bb5226ec573a2b0390679aebb/src/overlays/actors/ovl_player_actor/z_player.c#L3618):
```c
        } else if ((func_80124110(this, Player_ItemToActionParam(this, item)) != 0) && (gSaveContext.jinxTimer != 0)) {
            if (Message_GetState(&play->msgCtx) == TEXT_STATE_NONE) {
                Message_StartTextbox(play, 0xF7, NULL);
            }
        } else {
```


In [player](https://github.com/Decompollaborate/mm/blob/65f81484abbaa38bb5226ec573a2b0390679aebb/src/overlays/actors/ovl_player_actor/z_player.c#L3576):
(the first function called)
```c

    if (((this->actor.id == ACTOR_PLAYER) && (this->itemAction >= PLAYER_IA_FISHING_ROD)) &&
        !(((func_80124148(this) == 0) || (gSaveContext.jinxTimer == 0)) &&
          (func_8082FC78(this, (IREG(1) != 0) ? ITEM_FISHING_ROD : Inventory_GetBtnBItem(play)) ||
           func_8082FC78(this, C_BTN_ITEM(EQUIP_SLOT_C_LEFT)) || func_8082FC78(this, C_BTN_ITEM(EQUIP_SLOT_C_DOWN)) ||
           func_8082FC78(this, C_BTN_ITEM(EQUIP_SLOT_C_RIGHT))))) {
        func_80831990(play, this, ITEM_NONE);
    } else {
```